### PR TITLE
[IMP] mail: improve channel member list

### PR DESCRIPTION
**Pull request by seb-odoo**

### CODE REVIEW

File: addons/mail/static/src/components/thread_view/thread_view.xml [hunk 1]
```diff
@@ -8,57 +8,65 @@
...
-                        isCompact="(threadView.thread.model === 'mail.channel' and threadView.thread.mass_mailing) ? false : undefined"
-                        isDoFocus="props.isDoFocus"
-                        showAttachmentsExtensions="props.showComposerAttachmentsExtensions"
-                        showAttachmentsFilenames="props.showComposerAttachmentsFilenames"
-                        textInputSendShortcuts="threadView.textInputSendShortcuts"
-                        t-ref="composer"
-                    />
-                </t>
+                    <t t-if="threadView.thread and threadView.thread.model === 'mail.channel' and threadView.hasMemberList">
+                        <div class="o_ThreadView_coreChannelMemberListSeparator"/>
```
Comment (alexkuhn) [inline addons/mail/static/src/components/thread_view/thread_view.xml:66]: Would remove the prefix "core" in the part name, it's doesn't look important concern and resulting name would be easier to catch (feels less technical)

File: addons/mail/static/src/models/thread_view/thread_view.js [hunk 2]
```diff
@@ -237,6 +265,11 @@ function factory(dependencies) {
         hasComposerFocus: attr({
             related: 'composer.hasFocus',
         }),
+        /**
+         * Determines whether this thread view should show its member list.
+         * Only makes sense if this thread is a channel.
+         */
+        hasMemberList: attr({}),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/thread_view/thread_view.js]: Don't think you need empty `{}`

File: addons/mail/static/src/components/thread_view/thread_view.xml (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/thread_view/thread_view.xml:66]: This is the separator between "core" and "member list" so I prefer to keep both names in there (but I agree it's a bit verbose)

File: addons/mail/static/src/models/thread_view/thread_view.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/thread_view/thread_view.js]: I think I need to add a default there actually, I'll add true for now.

**Pull request merged**